### ISSUE TITLE
Message box crashing?

### ISSUE THREAD

**Issue by SimplisticApps** [chat-sdk/chat-sdk-android#35]
Hello again,

When i first used and setup the audio module, everything worked as it should but after creating a new project and importing and setting everything up again, it has started to crash for some reason?

The log:   `03-06 18:53:48.923 2011-2011/com.skintmedia.voice E/AndroidRuntime: FATAL EXCEPTION: main
                                                                    Process: com.skintmedia.voice, PID: 2011
                                                                    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.skintmedia.voice/com.braunster.chatsdk.activities.ChatSDKLoginActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.braunster.chatsdk.network.AbstractNetworkAdapter.facebookEnabled()' on a null object reference
                                                                        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2434)
                                                                        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2494)
                                                                        at android.app.ActivityThread.access$900(ActivityThread.java:157)
                                                                        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1356)
                                                                        at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                        at android.os.Looper.loop(Looper.java:148)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5551)
                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:730)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:620)
                                                                     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.braunster.chatsdk.network.AbstractNetworkAdapter.facebookEnabled()' on a null object reference
                                                                        at com.braunster.chatsdk.activities.ChatSDKLoginActivity.onCreate(ChatSDKLoginActivity.java:44)
                                                                        at android.app.Activity.performCreate(Activity.java:6272)
                                                                        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
                                                                        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2387)
                                                                        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2494) 
                                                                        at android.app.ActivityThread.access$900(ActivityThread.java:157) 
                                                                        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1356) 
                                                                        at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                        at android.os.Looper.loop(Looper.java:148) 
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5551) 
                                                                        at java.lang.reflect.Method.invoke(Native Method) 
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:730) 
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:620) `

Line 52:  `<com.braunster.chatsdk.view.ChatMessageBoxView
        android:layout_width="fill_parent"
        android:layout_height="wrap_content"
       52>>   android:id="@+id/chat_sdk_message_box"
       </com.braunster.chatsdk.view.ChatMessageBoxView>`

Thanks

**Comment by simonsmiley64** [comment]
Hi @SimplisticApps,

Can you confirm when the crashing occurred? For example when you changed the Firebase details of the audio project did it still occur or only after you had added it to your project?

I also noticed the similar text in the error message talking about the `facebookEnabled()` function. It might be worth removing this line if you are not using Facebook login. If you have not configured Facebook it might be causing issues.

Let me know about where the errors occur (Firebase configuration or adding the chat to your project) and I'll help narrow down the issue as best I can.

Simon

**Comment by SimplisticApps** [comment]
It looks like i put the wrong log in the comment! 

This is what comes up when i click on the user to open up the chat activity:

`03-07 14:43:26.820 16806-16806/com.skintmedia.voice E/AndroidRuntime: FATAL EXCEPTION: main
                                                                      Process: com.skintmedia.voice, PID: 16806
                                                                      java.lang.RuntimeException: Unable to start activity ComponentInfo{com.skintmedia.voice/com.braunster.chatsdk.activities.ChatSDKChatActivity}: android.view.InflateException: Binary XML file line #53: setAudioSource failed.
                                                                          at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2434)
                                                                          at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2494)
                                                                          at android.app.ActivityThread.access$900(ActivityThread.java:157)
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1356)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                          at android.os.Looper.loop(Looper.java:148)
                                                                          at android.app.ActivityThread.main(ActivityThread.java:5551)
                                                                          at java.lang.reflect.Method.invoke(Native Method)
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:730)
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:620)
                                                                       Caused by: android.view.InflateException: Binary XML file line #53: setAudioSource failed.
                                                                          at android.view.LayoutInflater.inflate(LayoutInflater.java:539)
                                                                          at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
                                                                          at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
                                                                          at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:409)
                                                                          at android.app.Activity.setContentView(Activity.java:2171)
                                                                          at com.braunster.chatsdk.activities.ChatSDKChatActivity.onCreate(ChatSDKChatActivity.java:30)
                                                                          at android.app.Activity.performCreate(Activity.java:6272)
                                                                          at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
                                                                          at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2387)
                                                                          at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2494) 
                                                                          at android.app.ActivityThread.access$900(ActivityThread.java:157) 
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1356) 
                                                                          at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                          at android.os.Looper.loop(Looper.java:148) 
                                                                          at android.app.ActivityThread.main(ActivityThread.java:5551) 
                                                                          at java.lang.reflect.Method.invoke(Native Method) 
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:730) 
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:620) 
                                                                       Caused by: java.lang.RuntimeException: setAudioSource failed.
                                                                          at android.media.MediaRecorder.setAudioSource(Native Method)
                                                                          at com.braunster.chatsdk.view.ChatMessageBoxView.setupRecorder(ChatMessageBoxView.java:286)
                                                                          at com.braunster.chatsdk.view.ChatMessageBoxView.onFinishInflate(ChatMessageBoxView.java:103)
                                                                          at android.view.LayoutInflater.rInflate(LayoutInflater.java:844)
                                                                          at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798)
                                                                          at android.view.LayoutInflater.rInflate(LayoutInflater.java:838)
                                                                          at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798)
                                                                          at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
                                                                          at android.view.LayoutInflater.inflate(LayoutInflater.java:423) 
                                                                          at android.view.LayoutInflater.inflate(LayoutInflater.java:374) 
                                                                          at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:409) 
                                                                          at android.app.Activity.setContentView(Activity.java:2171) 
                                                                          at com.braunster.chatsdk.activities.ChatSDKChatActivity.onCreate(ChatSDKChatActivity.java:30) 
                                                                          at android.app.Activity.performCreate(Activity.java:6272) 
                                                                          at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108) 
                                                                          at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2387) 
                                                                          at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2494) 
                                                                          at android.app.ActivityThread.access$900(ActivityThread.java:157) 
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1356) 
                                                                          at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                          at android.os.Looper.loop(Looper.java:148) 
                                                                          at android.app.ActivityThread.main(ActivityThread.java:5551) 
                                                                          at java.lang.reflect.Method.invoke(Native Method) 
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:730) 
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:620) 
`

I'm still using the stock code with my own keys but it won't work but it does work when i use the keys and setup straight from the .git included in the download you emailed?

Thanks

**Comment by SimplisticApps** [comment]
Instead of opening another issue, i was wondering if you could also tell how to see if an email address exists when you send a password change email because i don't want any emails being sent an address that doesn't exist on my firebase

Thanks

**Comment by SimplisticApps** [comment]
Update: I've had to work around the error i had so I'm okay with that now, i just need to know how to delete contacts and how to see if an email exists on the database or not

**Comment by kykrueger** [comment]
Sorry about the delays! Please open another issue if the email confirmations are still an issue. However, Firebase handles most of the user authentication.
